### PR TITLE
Seamless scraper

### PR DESCRIPTION
**Pull request by Anko59**

### CODE REVIEW

File: aiden-recommender/aiden_recommender/main.py [hunk 1]
```diff
@@ -18,13 +18,22 @@ class JobOfferRequest(BaseModel):
...
     job_reference: str
 
 
+class ScrapeResponse(BaseModel):
+    status: str
+    scrape_id: UUID
+
+
+class ScrapeStatusResponse(BaseModel):
+    status: str
```
Comment (Smirkey) [inline aiden-recommender/aiden_recommender/main.py:34]:
```suggestion
    status: ScrapeStatus
```

File: aiden-recommender/aiden_recommender/scrapers/scraper_aggregator.py [hunk 2]
```diff
@@ -103,5 +104,30 @@ async def search_jobs(self, search_query: str, location: str, profile_embedding_
...
+
+        logger.warning("Waiting for results")
+        try:
+            async with self.condition:
+                await asyncio.wait_for(
+                    self.condition.wait_for(lambda: self.request_queue.empty() and self.active_workers._value == self.workers),
+                    timeout=self.timeout,
+                )
+        except asyncio.TimeoutError:
+            logger.warning("Timeout reached")
```
Comment (Smirkey) [inline aiden-recommender/aiden_recommender/scrapers/scraper_aggregator.py:123]:
```suggestion
            logger.error("Timeout reached")
```

File: aiden-shared/aiden_shared/models.py [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
 from typing import Optional
 from datetime import datetime
 
 from pydantic import BaseModel
 
 
+class ScrapeStatus(Enum):
+    IN_PROGRESS = "in_progress"
+    FINISHED = "finished"
+    NOT_FOUND = "not_found"
```
Comment (Smirkey) [inline aiden-shared/aiden_shared/models.py:12]:
```suggestion
class ScrapeStatus(Enum):
    IN_PROGRESS = auto()
    FINISHED = auto()
    NOT_FOUND = auto()
```

File: aiden-shared/aiden_shared/models.py [hunk 4]
```diff
@@ -1,10 +1,17 @@
 from __future__ import annotations
+from enum import Enum
```
Comment (Smirkey) [inline aiden-shared/aiden_shared/models.py:2]:
```suggestion
from enum import Enum, auto
```

File: aiden-web/aiden_app/services/chat_service.py [hunk 5]
```diff
@@ -40,36 +39,16 @@ def get_profile_from_session(session) -> UserProfile:
...
-                    "role": role,
-                    "content": content,
-                }
-                yield render_to_string("langui/message.html", response)
+                message: ToolMessage = message
+                yield AssistantMesssage(
+                    title=message.function_nane,
+                    content=message.user_message,
+                    container_id=message.container_id,
+                ).model_dump_json()
```
Comment (Smirkey) [inline aiden-web/aiden_app/services/chat_service.py:53]: :relieved:

File: aiden-web/aiden_app/services/tools/scraper_tool.py [hunk 6]
```diff
@@ -1,29 +1,116 @@
...
+        self.returned_offers[scrape_id].append(job_offer.reference)
+
+        return job_offer
+
+    def format_user_message(self, job_offer: JobOffer) -> str:
+        if job_offer.profile is not None:
+            job_offer.profile = markdown2.markdown(job_offer.profile)
+        if job_offer.organization.description is not None:
+            job_offer.organization.description = markdown2.markdown(job_offer.organization.description)
+        return render_to_string("langui/job-offer.html", {"job_offer": job_offer.model_dump()})
```
Comment (Smirkey) [inline aiden-web/aiden_app/services/tools/scraper_tool.py:72]: this is the magic behind solving html escapes problem ?

**Review by Smirkey — APPROVED** [review]
really good, haven't tried locally, but a few suggestions

File: aiden-web/aiden_app/services/tools/scraper_tool.py (hunk 6 above)
Comment (Anko59) [reply aiden-web/aiden_app/services/tools/scraper_tool.py:72]: Nope, the magic is just sending offers one by one. This is the same parsing that we did before but in a cleaner place.

**Pull request merged**